CDSE SEN2 parcels conversion

jzvolensky opened the pull request:
Conversion progress

- [X]  Check data availability and loading
- [X] Filter and combine zarr products into a single xarray object
- [x] Reproduce OpenEO Apply kernel for dilation
- [x] Copy and adapt UDF segmentation code from [here](https://github.com/Open-EO/openeo-community-examples/tree/main/python/ParcelDelineation)
- [x] Download and apply ONNX ML model
- [x] Verify results

clausmichele commented:
@jzvolensky you can check how @annefou opened multiple Zarr objects here: https://eopf-sample-service.github.io/eopf-sample-webinar2/Webinar2-EOPF-Sample-Service-Multiple-files.html

clausmichele commented:
@jzvolensky `apply_kernel` implementation is here, it's already in Xarray: https://github.com/Open-EO/openeo-processes-dask/blob/181b7b56db97464e967a7cf1767ee64aad0847e0/openeo_processes_dask/process_implementations/cubes/apply.py#L131

jzvolensky commented:
Current status:

- Fixed zarr data filtering to match the temporal and spatial extent properly
- Updated masking
- Calculation of NDVI using properly masked data

The NDVI calculation should be aligned with what the ML models expect.
- Currently looking into how to reproduce the OpenEO ML inference step

clausmichele commented:
@jzvolensky can you please change the target branch to `dev`?
